src/lib/connect/socks.ts

```diff
@@ -0,0 +1,155 @@
+import _debug from 'debug'
+import duplexify, { Duplexify } from 'duplexify'
+import { SocksClient, SocksProxy } from 'socks'
+import { lookup } from 'dns'
+import { SocksProxyType } from 'socks/typings/common/constants'
+import { IStream } from '../shared'
+import { promisify } from 'util'
```

Comment:
I refactored the code a lot in order to drop duplexify, I suggest you to use BufferedStream util class instead If the goal is to buffer requests until connection

Reply:
Fair enough. The reason why I used it is to be able to immediately return the stream even though the actual socket only becomes available when the proxy has connected (so, indeed, buffering 😏 ).

This merge request is based on an older version that I did against an older version of MQTT.js which had duplexify, so I went with what was there back then, and when I rewrote it I didn't think much about it and just pulled it back in.

That said, I can definitely drops duplexify for any other way to achieve the same means. I am not sure what you refer to by `BufferedStream` though --- there's `BufferedDuplex`, but that's very specific to web socket connections. I can roll a similar helper based on nodejs `stream.Duplex` though if that's what you prefer.

Reply:
Done.